src/components/VerifiedFieldComponents.tsx

```diff
@@ -0,0 +1,174 @@
+  const [isFocused, setIsFocused] = useState(false);
+
+  const valuePath = `${path}.value`;
+  const verifiedPath = `${path}.verified`;
+  const verified: boolean = useWatch({
+    control,
+    name: verifiedPath,
+  });
+
+  return (
+    <Box
+      className={`flex items-center p-1 border-2 rounded-tr-md rounded-br-md ${
+        isFocused ? "border-fertiscan-blue" : ""
+      } ${className}`}
+      data-testid={`verified-field-${path}`}
+    >
+      <Typography
+        className="px-2 !font-bold select-none"
+        data-testid={`field-label-${path}`}
+      >
+        {label}
+      </Typography>
+      {renderField({ setIsFocused, control, valuePath, verified })}
+      <Divider
+        orientation="vertical"
+        flexItem
+        className={isFocused ? "!border-fertiscan-blue" : ""}
+        data-testid={`divider-${path}`}
+      />
+      <Controller
+        name={verifiedPath}
+        control={control}
+        render={({ field: { value, onChange } }) => (
+          <Tooltip
+            title={
+              verified
+                ? t("verifiedField.unverify", { label })
+                : t("verifiedField.verify", { label })
+            }
+            enterDelay={1000}
+          >
+            <IconButton
+              onClick={() => onChange(!value)}
+              data-testid={`toggle-verified-btn-${verifiedPath}`}
+              aria-label={
+                verified
+                  ? t("verifiedField.unverify", { label })
+                  : t("verifiedField.verify", { label })
+              }
+            >
+              <CheckIcon
+                className={value ? "text-green-500" : ""}
+                data-testid={`verified-icon-${verifiedPath}`}
+                aria-hidden
+              />
+            </IconButton>
+          </Tooltip>
+        )}
+      />
+    </Box>
+  );
+}
+
+function VerifiedCheckbox({
+  label,
+  path,
+  className = "",
+}: {
+  label: string;
+  path: string;
+  className?: string;
+}) {
+  const { t } = useTranslation("labelDataValidationPage");
+  return (
+    <VerifiedFieldWrapper
+      label={label}
+      path={path}
+      className={className}
+      renderField={({ setIsFocused, control, valuePath, verified }) => (
+        <Controller
+          name={valuePath}
+          control={control}
+          render={({ field }) => (
+            <Checkbox
+              {...field}
+              checked={field.value}
+              disabled={verified}
+              onFocus={() => setIsFocused(true)}
+              onBlur={() => setIsFocused(false)}
+              data-testid={`checkbox-field-${valuePath}`}
+              aria-label={`${t("verifiedInput.accessibility.checkbox", { label })}`}
+            />
+          )}
+        />
+      )}
+    />
+  );
+}
+
+function VerifiedInput({
+  label,
+  placeholder,
+  path,
+  className = "",
+}: {
+  label: string;
+  placeholder: string;
+  path: string;
+  className?: string;
+}) {
```

Comment:
Use arrow function syntax

Define an interface instead of having it inline for readability / consistency with other functions

Reply:
done